resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/RateLimiter.java

```diff
@@ -185,7 +191,14 @@ static RateLimiter ofDefaults(String name) {
     ) {
         return () -> {
             waitForPermission(rateLimiter, permits);
-            return supplier.get();
+            try {
+                F result = supplier.get();
+                drainIfNeeded(rateLimiter, CallsResult.success(result));
+                return result;
+            } catch (Exception exception) {
+                drainIfNeeded(rateLimiter, CallsResult.failure(exception));
```

Comment:
Please introduce and use an `onError` hook

Reply:
done